### DISCUSSION TITLE
[Enhancement request] Update dashboard variables from panel data links URL's

### DISCUSSION THREAD

**Discussion by bhozar** [grafana/grafana#38353] [Dashboards, Panels & Visualizations]
**What would you like to be added**:
Support updating variables in dashboard from Data Links in panels in a way which supports templating dashboards.

Example:
![image](https://user-images.githubusercontent.com/1184220/70921325-ff70a680-201b-11ea-9ac2-77d2a00b89a4.png)

**Why is this needed**:
I can put a link like above, but if I export the dashboard for Grafana.com it will break when imported into another Grafana system as the URL may be different.

**Comment by dprokop** [comment]
@bhozar thanks for reporting this. What do you mean by break? Not sure I understand your needs here - since your link is using variable it's by it's nature subject of a change. If you import to  a system in which the template variable values are different, then I cannot see a way of assuring the link is still valid.

**Comment by bhozar** [comment]
@dprokop If it supported a variable to reference the URL of the board you could include self-referencing variables in a dashboard I export. e.g the URL above could be displayed as:
`${selfurl}` and it would be replaced with `/grafana/d/VxymKw4Zz/bitvise-sftp-logs` or whatever the board URL on the imported system. That would mean that URL field would always work, even when exported/imported into another system.

Would also be helpful if we could add a variable for all existing variables, so it can take the current ones on the board and just add the additional one in place. Means you can have a single dashboard and filter your data down using the variables, similar to how Azure Application Insights/Log Analytics and Kibana can filter by clicking data points.

**Comment by bhozar** [comment]
The use case for this is: I would like to include links in the panels to filter on the data, but if the board name/URL changes it breaks.

This is most noticeable if I export the dashboard so I can upload it to Grafana.com as a template as it breaks all of them.

**Comment by torkelo** [comment]
you can use ${__dashboard.uid} to get the uid of the dashboard

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
